terraform/modules/coreos_asg/main.tf

```diff
@@ -0,0 +1,223 @@
+data "aws_region" "selected" {
+    current = true
+}
+
+data "aws_caller_identity" "current" {}
+
+resource "aws_ecs_cluster" "cluster" {
+    name = "${var.name}"
+}
+
+data "aws_ami" "coreos_stable" {
```

Comment:
This is not a k8s cluster. This is a standalone single-server solution that will manage the k8s cluster, terraform, automated testing etc.

If you want to write and maintian your own standalone baseimage that provides the ability to fully bootstrap everything I need via user-data I am happy to use that. We could also just switch to your drop-in replacement later once it exists with a 5 line PR here.

In this case being super minimal is pretty ideal imo.